### PR TITLE
Remove low-level log functions.

### LINKED ISSUE #4295: Remove low-level log functions
I think they are not really used (events are much more flexible) and if people actually do want a low-level interface, they can use assembly.

### PR DESCRIPTION
**Pull request by chriseth**
Fixes #4295

### CODE REVIEW

File: test/libsolidity/SolidityEndToEndTest.cpp [hunk 1]
```diff
@@ -1177,139 +1177,6 @@ BOOST_AUTO_TEST_CASE(uncalled_blockhash)
...
-		BOOST_REQUIRE_EQUAL(numLogs(), 1);
-		BOOST_CHECK_EQUAL(logAddress(0), m_contractAddress);
-		BOOST_CHECK_EQUAL(h256(logData(0)), h256(u256(1)));
-		BOOST_REQUIRE_EQUAL(numLogTopics(0), 4);
-		for (unsigned i = 0; i < 4; ++i)
-			BOOST_CHECK_EQUAL(logTopic(0, i), h256(u256(i + 2)));
-	)
-}
-
-BOOST_AUTO_TEST_CASE(log_in_constructor)
```
Comment (axic) [inline test/libsolidity/SolidityEndToEndTest.cpp:1292]: Do we have this test case with `event`?

Comment (chriseth) [reply test/libsolidity/SolidityEndToEndTest.cpp:1292]: Added a test that uses an event in the constructor: https://github.com/ethereum/solidity/pull/10092/files#diff-452c3979dc6e8b81f702d5c88aa872017141cf6a32d1f2ea839ac6750a735127R1812

File: docs/080-breaking-changes.rst [hunk 2]
```diff
@@ -17,3 +17,10 @@ and it does something else afterwards.
 * Explicit conversions from negative literals and literals larger than ``type(uint160).max`` to ``address`` are now disallowed.
 * Exponentiation is right associative, i.e., the expression ``a**b**c`` is parsed as ``a**(b**c)``.
   Before 0.8.0, it was parsed as ``(a**b)**c``.
+
+Syntactic Only Changes
```
Comment (axic) [inline docs/080-breaking-changes.rst:21]: "Syntactic Changes", "Syntax Only Changes", or "Only Syntactic Changes" I think are more correct.

**Review by axic — APPROVED** [review]
Looks good, but please consider the text suggestion.

File: docs/080-breaking-changes.rst (hunk 2 above)
Comment (chriseth) [reply docs/080-breaking-changes.rst:21]: We have to re-group them somehow. It's also not really a "syntax only change"

**Pull request merged**